### DISCUSSION TITLE
StateMap keepalive?

### DISCUSSION THREAD

**Discussion by mhite** [icedream/go-stagelinq#16] [Ideas]
Curious if you've seen any evidence of a StateMap "keepalive" feature to detect dead connections on the client side? I've seen instances where the Denon stops sending updates to the subscriber, yet the subscriber (go-stagelinq) and OS consider the TCP connection up so it continues to obediently wait for updates.

**Comment by icedream** [comment]
As of right now, I can only suspect that I didn't record enough traffic to observe other messages being sent from the client side to keep the connection up. So, not that I would know of at least.

**Comment by mhite** [comment]
No problem, just was curious if you've seen evidence of how client and/or server deal with dead peers in the StateMap service. Thanks, and great job on the reverse engineering work BTW!

**Comment by honusz** [comment]
I suspect there may be some keepalive functionality with the TimeSync service; well-behaving StageLinq devices seem to send these messages back and forth. 
That being said, I don't recall any situations where StateMap stopped updating while the TCP connection was still open, and I've left test sessions open overnight accidentally and when I touched the platter, it still broadcast the message.

I haven't gotten TimeSync to work when we initiate the connection to the device, only in some tests where I got the device to initiate the connection to us (you have to use a token that more resembles a SoundSwitch token, where the first several bytes are 00).

**Comment by chrisle** [comment]
Interestingly, I'm just running into the same problem as @mhite.

**Comment by honusz** [comment]
@chrisle @mhite which PrimeOS firmware version?

**Comment by chrisle** [comment]
@honusz Prime OS 2.2.2

**Comment by mhite** [comment]
@honusz I'm also running 2.2.2
